pintc/src/expr/evaluate.rs

```diff
@@ -503,12 +503,50 @@ impl Evaluator {
                     }
                 }),
 
+            Expr::In {
+                value, collection, ..
+            } => {
+                let value = self.evaluate_key(value, handler, contract)?;
+
+                if let Expr::Range { lb, ub, .. } = collection.get(contract) {
+                    let lb = self.evaluate_key(lb, handler, contract)?;
+                    let ub = self.evaluate_key(ub, handler, contract)?;
+
+                    match (lb, ub, value) {
+                        (Imm::Int(lb), Imm::Int(ub), Imm::Int(value)) => {
+                            Ok(Imm::Bool(ub >= value && lb <= value))
+                        }
+
+                        _ => Err(handler.emit_err(Error::Compile {
+                            error: CompileError::Internal {
+                                msg: "unexpected expression during compile-time evaluation \
+                            evaluation of in expr with range",
+                                span: empty_span(),
+                            },
+                        })),
```

Comment:
My understanding is that we only support `ranges` with `ints` and that `reals` are experimental. If that is the case, then it should be fine to only evaluate `in exprs` structured as `int in int..int`?

Also I'll be updating the error messages, as I don't believe they should be internal. A user can still write `const f = true in 3..2;` without an error occurring before this stage.

FWIW, if it not a `const` then we type check the range. Ex.
```
predicate Test {
    var l = true in 1..2;
}

Error: value type and range type differ
    ╭─[test.pnt:18:21]
    │
 18 │     var l = true in 1..2;
    │                     ┬
    │                     ╰── range type mismatch; expecting `bool` type, found `int` type
────╯
```

Reply:
1. We should support `real` regardless since we do handle it in other parts of the evaluator. Should be fairly straightofward. To test it, add a test under `pintc/tests/experimental/` and run with `cargo test --all-features`.
2. Yeah keep the internal error for now but open an issue to error out properly in the type checker.